File: c_solutions_1-10/Euler_5.c
#include <stdio.h>
#include <time.h>


int divide_evenly(long dividend) {

    int i;
    for (i=1; i <= 20; i++) {
        if (dividend % i != 0) 
            return 0;
    }
    return dividend;
}

int main(void) {

    clock_t start, stop;
    start = clock();

    long count;
    for (count=20; !(divide_evenly(count)); count+=20)
        ;

    stop = clock();    
    printf ("Answer: %ld\n", count);
    printf ("Time: %f\n", ((float)stop - (float)start) / CLOCKS_PER_SEC);
    return 0;
}

File: perl_solutions_1-10/Euler_5.pl
use strict;
use warnings;
use Time::HiRes qw( clock );


sub range_divisor
{
    my ($low, $high) = @_;
    my $start = $high;
    while (1) {
        foreach my $i ($low..$high) {
            if ($start % $i != 0) {
                $start += $high;
                last;
            }
            elsif ($i == $high) {
                return $start;
            }
        }
    }
}

my $start = clock();
my $answer = range_divisor(1, 20);
my $stop = clock();
print "Answer: ", $answer, "\n";
printf "Time: %.5f\n", $stop - $start;
